### DISCUSSION TITLE
Help with reducing time difference between event and alert

### DISCUSSION THREAD

**Discussion by lymanaukuleles** [jertel/elastalert2#824] [Q&A] [answered]
My log that has the query I want to alert on is indexed into Elasticsearch and viewable in Kibana. However my ElastAlert alerts on this event about two hours after it is viewable in Kibana. This has only been happening the last couple of weeks. Before then, I would get the alert almost immediately. 

Any idea how I can fix this?

**Comment by jertel** [comment, marked as answer]
We can't offer a solution without knowing the problem, and all we currently have is the symptom. You stated the problem started recently, but used to work. You've not mentioned any changes to your ElastAlert 2 installation or your Elasticsearch cluster during that same period. So I have to assume that nothing changed, and the symptoms began occurring on their own. Therefore to find the underlying problem you will need to watch the elastalert 2 logs, and enable debug logging if necessary, to track down what is going on. It could be that your ElastAlert 2 is temporarily hanging, or taking a very long time to query the data. Watching the logs and seeing the results coming back from each ElastAlert 2 query to ES should help you.

**Reply by lymanaukuleles** [reply]
I think it occurred after doing security updates on my Linux VM. That might not be enough info though. I was thinking about completely reinstalling Docker (which is how I'm using ElastAlert2) because I have another random issue with it. The `grep `command is no longer working on my `docker logs` for some reason (it used to work fine). What I mean is when I run `grep` with the `docker logs` command it will just print out everything, and not target the specific string. I have no idea why this is occurring but maybe a reinstall of Docker will fix things? Also the `grep` command not working was before doing the security updates.

**Reply by jertel** [reply]
If you run `docker logs elastalert | grep "something unique"` it still dumps everything to your screen? I've never seen grep fail, and I don't think I've ever had to re-install Docker before. So that's not something I would personally be considering as the culprit. But stranger things have happened!

**Reply by lymanaukuleles** [reply]
Yes everything is dumped to the screen when prior it would select only the specified string. Also my grep works fine for other operations on the VM, it is only on Docker where it fails.

**Reply by jertel** [reply]
Perhaps the ElastAlert 2 container is outputting its logs to stderr instead of stdout. Try using `docker logs elastalert 2>&1 | grep "something unique"`.

**Reply by lymanaukuleles** [reply]
That's the solution ^ !

stderr must be for --debug mode and stdout must be for --verbose mode
